### DISCUSSION TITLE
Reflection Probes could be Improved

### DISCUSSION THREAD

**Discussion by rhodochrone** [godotengine/godot-proposals#6723] [3D]
![image](https://user-images.githubusercontent.com/75505811/233348146-7f0cf99b-70d5-4cb8-804a-ea378a1cc431.png)

Whether or not you enable "Interior", the rendering of ReflectionProbe cubemaps makes no sense to me. I think the Probe should simply represent the contents of the cubemap you render. Instead you get two modes, one where the sky is layered on top of the reflection  (which already renders the sky appropriately, thus making the rest of the geometry in the reflection appear transparent?), and another which simply ignores the sky entirely, which also doesn't make sense, because a cubemap in an interior will have a naturally occluded sky due to the geometry in the way. Interiors often have windows and these will just look wrong. I can't see any reason for cubemaps not to render the sky if it's visible from the probe's rendering perspective.

![image](https://user-images.githubusercontent.com/75505811/233352064-07427828-5785-45b6-88e2-865bfc613fcc.png)

Do you suppose it would be reasonable to have a two-layered reflection when Box Projection is enabled? I suppose transparent geometry might complicate this, but if the sky is rendered without Box Projection first, then the Box Projected cubemap is layered on top with the sky "cut out", you'd get a more accurate representation of the scene's parallax. 

On another note, it would be great to have MSAA support for the probes, especially if the probes are just static and don't need to update in real time.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/3013 and 
https://github.com/godotengine/godot/issues/49789.

> one where the sky is layered on top of the reflection (which already renders the sky appropriately, thus making the rest of the geometry in the reflection appear transparent?),

This is because the probe fades towards the end of its size. This allows the probe to blend with parts of the scene that lack reflection probe coverage. Increase the probe's size to reduce this effect until https://github.com/godotengine/godot/pull/61416 is merged (so you can adjust this effect).
